### ISSUE TITLE
Axis labels for NA-Data with factor levels

### ISSUE THREAD

**Issue by fabiangehring** [ropensci/plotly#910]
When having a NA-Dataframe, but correct facor levels for x and y, the created empty plot does not label the axis according to the factor levels. 

`data <- data.frame(a = factor(character(), levels = c("Entry1", "Entry2")), b = factor(numeric(), levels = c(1, 2)))
plot_ly(data, x = ~a, y = ~b, type = "bar")`

**Comment by cpsievert** [comment]
I agree this is somewhat surprising, but do you have a practical example? You want to plot axes, but no data?

**Comment by fabiangehring** [comment]
I think my examples are a bit misleading. In my shiny app I often Encounter situations where no data is available for known categories, i.e.

```
data <- data.frame(a = c("a", "b"), b = NA_real_)
plot_ly(data, x = ~a, y = ~b, type = "bar")
```

```
data <- data.frame(a = factor(c("a", "b"), c("a", "b", "c"), b = NA_real_)
plot_ly(data, x = ~a, y = ~b, type = "bar")
```

I know that I could use 0 instead of NA for plotting in such situations and the pl. But since I am exporting the data to csv as well I want them to be NA. I therefore think it is natural

**Comment by cpsievert** [comment]
Maybe you could just print an empty plot if the data is missing?

```r
plotly_empty(type = "bar")
```

It seems a bit strange to know the categories, but not the actual data. That being said, if you need to show the axes, you could hack up something like this:

```r
plot_ly() %>% layout(xaxis = list(type = "category", tickvals = 1:5/5, ticktext = letters[1:5], range = c(0, 1)))
```

**Comment by dracodoc** [comment]
@cpsievert , I'm having a similar question and I cannot make it work. Can you take a look [here](https://stackoverflow.com/questions/60779748/how-to-force-plotly-r-to-plot-missing-values-with-category-axis)?
